internal/ovirt/data_source_ovirt_wait_for_ip_test.go

```diff
@@ -0,0 +1,101 @@
+package ovirt
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	ovirtclient "github.com/ovirt/go-ovirt-client"
+)
+
+func TestOvirtWaitForIP(t *testing.T) {
+	t.Parallel()
+
+	_, err := os.Stat("./testimage/full.qcow")
+	if err != nil {
+		t.Skipf("Full test image is not available, please run go generate.")
+	}
+
+	p := newProvider(newTestLogger(t))
+	helper := p.getTestHelper()
+	client := helper.GetClient().WithContext(context.Background())
+	clusterID := helper.GetClusterID()
+	templateID := helper.GetBlankTemplateID()
+	vnicProfileID := helper.GetVNICProfileID()
+	storageDomainID := helper.GetStorageDomainID()
+	config := fmt.Sprintf(
+		`
+provider "ovirt" {
+	mock = true
+}
+
+resource "ovirt_vm" "foo" {
+	cluster_id = "%s"
+	template_id = "%s"
+	name = "test"
+}
+
+resource "ovirt_nic" "foo" {
+  name            = "eth0"
+  vm_id           = ovirt_vm.foo.id
+  vnic_profile_id = "%s"
+}
+
+resource "ovirt_disk_from_image" "foo" {
+	storage_domain_id = "%s"
+	format           = "cow"
+    alias            = "test"
+    sparse           = true
+    source_file      = "./testimage/full.qcow"
+}
+
+resource "ovirt_disk_attachment" "foo" {
+	vm_id          = ovirt_vm.foo.id
+	disk_id        = ovirt_disk_from_image.foo.id
+	disk_interface = "virtio_scsi"
+}
+
+resource "ovirt_vm_start" "foo" {
+	vm_id = ovirt_vm.foo.id
+
+	depends_on = [ovirt_nic.foo, ovirt_disk_attachment.foo] 
+}
+
+data "ovirt_wait_for_ip" "test" {
+    vm_id = ovirt_vm.foo.id
```

Comment:
Minor: Couldn't we use `ovirt_vm_start.foo.id` instead and omit the `depends_on` part as in the example?